fix(ci-mkdocs): Mkdocs cache invalidation

neo-jesse opened the pull request:
We're invalidating the Cloudfront cache at the end of the deploy, to resolve an issue where we get inconsistent results after deploy.

neo-jesse committed:
step to invalidate cache

neo-jesse committed:
Testing dist id

neo-jesse committed:
correct DIST ID

neo-jesse committed:
Using github secrets

y3rsh reviewed (changes requested):
Thanks 🙌  
We’ll want to handle the CloudFront cache invalidation logic inside the Python deploy script, right alongside the S3 sync. That way:  
- A user can run the deploy locally and everything (sync + invalidate) happens in one place.  
- We don’t have to split logic between the workflow and the script, which makes it harder to debug if something goes wrong.  
- We can reuse the existing configuration object and just add a mapping of the correct CloudFront distributions to invalidate.  

If we keep this inside the workflow, anyone troubleshooting or trying to deploy during a GitHub outage would have to look in multiple places to understand the deployment.

ncdiehl11 committed:
feat(protocol-designer, step-generation): allow no trash or waste chute in PD (#19375)

With the addition of return tip capabilities in PD, we want to remove
the requirement to have added a trash bin or waste chute. Any error
handling for command creators that require the presence of a trash bin
or waste chute entity will live inside the command creators themselves.

Closes AUTH-587

ncdiehl11 committed:
fix(protocol-designer): fix import modal close behavior (#19369)

`FileUploadMessagesModal` should force user action to acknowledge the
migration modal when importing an older protocol. This PR removes the
close button and close on click outside behavior

Closes AUTH-2234

neo-jesse committed:
testing cache inval. step in python

neo-jesse committed:
back to edge branch

neo-jesse commented:
✅ Successfully deployed to staging!
📍 Deployed to: https://staging.docs.opentrons.com/
Invalidating CloudFront cache for staging...
Running: aws cloudfront create-invalidation --distribution-id *** --paths /*
{
    "Location": "https://cloudfront.amazonaws.com/2020-05-31/distribution/***/invalidation/IAKRVQ3XNT821D6T4YMSXYIS00",
    "Invalidation": {
        "Id": "IAKRVQ3XNT821D6T4YMSXYIS00",
        "Status": "InProgress",
        "CreateTime": "2025-08-27T15:25:52.055000+00:00",
        "InvalidationBatch": {
            "Paths": {
                "Quantity": 1,
                "Items": [
                    "/*"
                ]
            },
            "CallerReference": "cli-1756308351-462532"
        }
    }
}

✅ CloudFront cache invalidation initiated for staging
   Invalidation ID: {
    "Location": "https://cloudfront.amazonaws.com/2020-05-31/distribution/***/invalidation/IAKRVQ3XNT821D6T4YMSXYIS00",
    "Invalidation": {
        "Id": "IAKRVQ3XNT821D6T4YMSXYIS00",
        "Status": "InProgress",
        "CreateTime": "2025-08-27T15:25:52.055000+00:00",
        "InvalidationBatch": {
            "Paths": {
                "Quantity": 1,
                "Items": [
                    "/*"
                ]
            },
            "CallerReference": "cli-1756308351-462532"
        }
    }
}

neo-jesse committed:
Merge branch 'edge' into mkdocs-cache-invalidation

neo-jesse commented:
STAGING_CLOUDFRONT_DISTRIBUTION_ID - Set this to E2DBE0K9VT8YB9 (or the correct staging distribution ID)
PRODUCTION_CLOUDFRONT_DISTRIBUTION_ID - Set this to E2PSPUXND1RQWG (or the correct production distribution ID)

neo-jesse commented:
> Thanks 🙌 We’ll want to handle the CloudFront cache invalidation logic inside the Python deploy script, right alongside the S3 sync. That way:
> 
>     * A user can run the deploy locally and everything (sync + invalidate) happens in one place.
> 
>     * We don’t have to split logic between the workflow and the script, which makes it harder to debug if something goes wrong.
> 
>     * We can reuse the existing configuration object and just add a mapping of the correct CloudFront distributions to invalidate.
> 
> 
> If we keep this inside the workflow, anyone troubleshooting or trying to deploy during a GitHub outage would have to look in multiple places to understand the deployment.

Invalidation step has been moved into the Python.
